src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -137,15 +137,28 @@ static bool findVersionInStream(std::istream& in,
 bool UnixOsParser::parseFile(std::istream& in, nlohmann::json& info)
 {
     constexpr auto SEPARATOR{'='};
-    static const std::map<std::string, std::string> KEY_MAP
+    bool ret;
+
+    static const std::map<std::string, std::string> KEY_MAP1
     {
         {"NAME",             "os_name"},
         {"VERSION",          "os_version"},
         {"ID",               "os_platform"},
         {"BUILD_ID",         "os_build"},
         {"VERSION_CODENAME", "os_codename"}
     };
```

Comment:
```suggestion
    static const std::map<std::string, std::string> KEY_MAP1
    {
        {"NAME",             "os_name"},
        {"VERSION_ID",       "os_version"},
        {"VERSION",          "os_version"},
        {"ID",               "os_platform"},
        {"BUILD_ID",         "os_build"},
        {"VERSION_CODENAME", "os_codename"}
    };
```

Reply:
The issue with this solution is that tags "VERSION_ID" and "VERSION" have no priority in case both are present in file os-release.
For example in Ubuntu both tags are present:

```
matias@matias-X580VD:~/Proyectos_Wazuh/wazuh/src$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```
Despite function "parseUnixFile" always takes the last tag ocurrence (in case of ubuntu would be "VERSION") and it would works. But in case that these tags were inverted (for example in other OS, I dont know), the current behaviour of the parser would be altered.
Thats why I choose to find in first place for tag "VERSION" and in case it not exists, then find for tag "VERSION_ID".

Reply:
As we agreed in the stand up, this implementation is going to be modified doing a rework of function parseUnixFile.